src/main/java/com/amplitude/Retry.java

```diff
@@ -177,6 +180,11 @@ private static void retryEventsOnLoop(String userId, String deviceId, HttpCall h
                     eventCount /= 2;
                   }
                 } catch (InterruptedException e) {
+                } catch (AmplitudeInvalidAPIKeyException e) {
+                  // The retry logic should only be executed after the API key checking passed.
+                  // This catch AmplitudeInvalidAPIKeyException is just for handling
+                  // retryEventsOnce in thread.
+                  Thread.currentThread().interrupt();
```

Comment:
looks fine, I guess return or interrupt both work. Do we also need this line for `InterruptedException`?

Reply:
Should we interrupt the thread or throw RuntimeException?
https://stackoverflow.com/a/22621931/11969222

Reply:
`AmplitudeInvalidAPIKeyException` is already an exception here, normally if we don't catch it should just stop thread from continue